superset/commands/database/create.py

```diff
@@ -118,25 +111,23 @@ def run(self) -> Model:
         ) as ex:
             event_logger.log_with_context(
                 action=f"db_creation_failed.{ex.__class__.__name__}",
-                engine=database.db_engine_spec.__name__,
```

Comment:
Here `database` can be undefined, raising an exception inside the catch block. Instead I compute the engine from the URI, and pass it to all instances where we log.